File: gameserver/src/main/java/l2s/gameserver/network/authcomm/gs2as/ReduceAccountPoints.java
package l2s.gameserver.network.authcomm.gs2as;

import io.netty.buffer.ByteBuf;
import l2s.gameserver.network.authcomm.SendablePacket;

public class ReduceAccountPoints extends SendablePacket {
    private final String account;
    private final int count;

    public ReduceAccountPoints(String account, int count) {
        this.account = account;
        this.count = count;
    }

    @Override
    protected ByteBuf writeImpl(ByteBuf byteBuf) {
        writeString(byteBuf, account);
        byteBuf.writeInt(count);
        return byteBuf;
    }

    @Override
    protected byte getOpCode() {
        return 18;
    }
}

File: gameserver/src/main/java/l2s/gameserver/network/authcomm/gs2as/ReduceAccountPoints.kt
package l2s.gameserver.network.authcomm.gs2as

import io.netty.buffer.ByteBuf
import l2s.gameserver.network.authcomm.SendablePacket

class ReduceAccountPoints(private val account: String, private val count: Int) : SendablePacket() {

    override val opCode: Byte = 0x12

    override fun writeImpl(byteBuf: ByteBuf): ByteBuf {
        writeString(byteBuf, account)
        byteBuf.writeInt(count)
        return byteBuf
    }

}